client/src/__test__/components.test.js

```diff
@@ -0,0 +1,92 @@
+import { configure, shallow, mount } from 'enzyme';
+import React from 'react';
+import Product from '../components/Product.jsx';
+import Price from '../components/Price.jsx';
+import Name from '../components/Name.jsx';
+import Styles from '../components/Styles.jsx';
+import Style from '../components/Style.jsx';
+
+
+import Adapter from 'enzyme-adapter-react-16';
+
+configure({ adapter: new Adapter() });
+
+describe('<ProductModule />', () => {
+  const details = {
+    name: 'Spring Nova X7',
+    gender: 'Men\'s',
+    category: 'Trail Running',
+    price: '670'
+  };
+  it('renders without crashing', () => {
+    shallow(<Product details={details} />);
+  });
+});
+
+describe('<Price />', () => {
+  it('renders without crashing', () => {
+    const price = '1200';
+    shallow(<Price price={price} />);
+  });
+  it('renders with a 0 price', () => {
+    const price = '0';
+    shallow(<Price price={price} />);
+  });
+  it('renders with an undefined price', () => {
+    const price = undefined;
+    shallow(<Price price={price} />);
+  });
+});
+
+describe('<Name />', () => {
+  it('renders without crashing', () => {
+    const name = 'Spring Nova X7';
+    shallow(<Name name={name}/>);
+  });
+});
+
+describe('<Styles />', () => {
+  const styles = [{
+    imageRegular: 'https://fake-url.com/1/001/regular.png',
+    imageThumbnail: 'https://fake-url.com/1/001/thumbnail.png',
+    options: 'Member Select',
+    price: '100',
+    styleId: '001'
+  },
+  {
+    imageRegular: 'https://fake-url.com/1/001/regular.png',
+    imageThumbnail: 'https://fake-url.com/1/001/thumbnail.png',
+    options: 'Member Select',
+    price: '200',
+    styleId: '002'
+  },
+  {
+    imageRegular: 'https://fake-url.com/1/001/regular.png',
+    imageThumbnail: 'https://fake-url.com/1/001/thumbnail.png',
+    options: 'Member Select',
+    price: '300',
+    styleId: '003'
+  }];
+  it('renders without crashing', () => {
+    shallow(<Styles styles={styles}/>);
+  });
+  it('renders 3 <Style /> components', () => {
```

Comment:
Your tests seem good to me.

However, one question I have is, why have them all in one file? Especially as you add more tests per component.

Besides testing for "renders without crashing," it would be a good idea to add more tests for presentation and state. (For example, you have here--testing the number of components rendered. You could also test if the rendered HTML matches the props, click events, responsive layout, etc.)

Reply:
These are all great suggestions.
- The components are very simple so I squished them in 1 file.  
- onHover, onClick will be added on a separate PR.